File: Problem2/p2.cpp
#include <iostream>
#include <vector>

int main(int argc, char **argv) {

    // Declare a vector to hold fib. numbers.
    std::vector<int> fibNumbers;
    std::vector<int> evens;
    fibNumbers.push_back(1);
    fibNumbers.push_back(2);
    int i = 3;
    int sum = 2;
    // Loop 4 million times.
    while (i < 4000000) {
        if ((i % 2) == 0) {
            evens.push_back(i);
            sum += i;
        }
        fibNumbers.push_back(i);
        i = (fibNumbers.end()[-2] + fibNumbers.end()[-1]);
    }

    std::cout << sum << std::endl;
    
    return 0;
}

File: Problem2/p2.py
"""
def fib(n):
    if n == 0:
        return 0
    if n == 1:
        return 1
    return fib(n-1) + fib(n-2)
go = True
total = 0
i = 2
while go:
    number = fib(i)
    if number > 4000000:
        go = False
    else:
        if (number % 2) == 0:
            total = total + number
        i = 1 + i
print total
"""
fibNumbers = [1, 2]
total = 2
nextNumber = 3

while not (nextNumber > 4000000):
    if nextNumber % 2 == 0:
        total = total + nextNumber
    fibNumbers.append(nextNumber)
    nextNumber = fibNumbers[-1] + fibNumbers[-2]

print total

